frontend/src/components/Modal/ConfirmModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import styles from "./ConfirmModal.module.scss";
+import { confirmUser } from "../../../API/UserService/confirmUser";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+import { useAppDispatch, useAppSelector } from "store/store";
+import { getProfilesForLoggedUser } from "store/profile";
+import { changeActiveProfile } from "store/activeProfile";
+
+interface ConfirmModalProps {
+    profileID: string | undefined;
+    changeVisiblity: () => void;
+    userID: string;
+    functionToConfirm: () => void;
+    toastName: string;
+}
+
+export const ConfirmModal = ({
+    changeVisiblity,
+    profileID,
+    userID,
+    functionToConfirm,
+    toastName,
+}: ConfirmModalProps) => {
+    const [password, setPassword] = useState("");
+    const [errorMessage, setErrorMessage] = useState(" ");
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+    const userProfiles = useAppSelector((state) => state.profiles.profiles);
```

Comment:
You can prepare this selector inside slice, so you dont have to pass state here: 
```tsx
export const selectProfiles = ({ profileSlice }: RootState) => {
  return profileSlice;
};